### PR TITLE
seq2seq.py: Fixed the documentation to be consistent with the code

### PR DESCRIPTION
**Pull request by sahiliitm**
A fix for https://github.com/tensorflow/tensorflow/issues/3800

### CODE REVIEW

**Comment by sahiliitm** [comment]
I signed it!

**Comment by lukaszkaiser** [comment]
Hi sahiliitm . I looked at your changed comments, but I don't understand them. What is `output_size`, what part of arguments is it? Are the current comments not correct? Can you explain?

**Comment by sahiliitm** [comment]
Hi Lukas,

I have documented the bug [here](https://github.com/tensorflow/tensorflow/issues/3800). Please let me know if it is not clear. :) 
In short, the outputs of the two functions (for which I have changed the comments) do not return tensors of size `(batch_size, num_decoder_symbols)` but rather of size `(batch_size, output_size)` where `output_size` is the same parameter as [documented for the function](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/seq2seq.py#L252) `embedding_rnn_decoder`

**Comment by lukaszkaiser** [comment]
Oh, I see, what is meant by output_size is cell.output_size (the comment just missed that). But I think you're not fully correct -- the functions do output num_decoder_symbol-sized tensors sometimes, it depends on the output_projection. If you're correcting it, please make it fully correct. I'll be away from github for a few weeks now, so there might be a delay in my answers, sorry.

**Comment by sahiliitm** [comment]
Are you sure that is the case? 
The reason I doubt that is because of this [line of code](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/models/rnn/translate/seq2seq_model.py#L161). We do pass the parameters `output_projection` to the function `tf.nn.seq2seq.model_with_buckets` (indirectly through the function defined at [this line](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/models/rnn/translate/seq2seq_model.py#L125)) in [seq2seq_model.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/models/rnn/translate/seq2seq_model.py) but the output returned by it is still a `cell.output_size` sized tensor. Which is why the line of code I pointed out is needed to project it back to the size `num_decoder_symbols`.

I guess what I'm asking is that if `output_projection` parameters are not `None`, is that the case when the functions under question would return `num_decoder_symbol` sized tensors? That does not seem to be the case.

**Comment by lukaszkaiser** [comment]
I think the seq2seq module behaves like you say in the end: if output_projection is not None, it returns cell.output_size - sized tensors. If it is None, it returns num_decoder_symbol - sized tensors. Is it not so?

The seq2seq_model example is a bit different. It uses sampled softmax, but that's only good for training. When decoding, it cannot sample from the softmax, so it just returns the dense logits. But that's seq2seq_model -- let's keep it separate from this issue, which is changing the seq2seq module.

Or am I missing something? Let me know!

**Comment by sahiliitm** [comment]
Hi Lukas, 

Sorry for the delayed reply. I was a bit busy. I agree with you. I've changed the comments accordingly.
thanks

**Comment by lukaszkaiser** [comment]
It looks good to me now, great thanks for clarifying and making the code better! There is one small nit: it looks like the indentation of your comments is off. I'm not even sure if that's the case of if it's just the online viewer, but could you take a look and make sure they're indented ok? (I mean: the multi-line argument comments should be a few spaces to the left -- same as other comments in the file.)

Except for this, we should merge this. Some checks seem to still be running...

**Comment by sahiliitm** [comment]
Hi,
Sorry! Indeed, some of the comments were badly formatted. I've fixed that now.
Thanks for pointing it out.

**Comment by lukaszkaiser** [comment]
Looks good to me, thanks! Can we merge this one? Thanks! (Assigned vrv@ just for merging.)

**Comment by vrv** [comment]
@tensorflow-jenkins test this please
